include/boost/regex/v4/mem_block_cache.hpp

```diff
@@ -19,17 +19,72 @@
 #define BOOST_REGEX_V4_MEM_BLOCK_CACHE_HPP
 
 #include <new>
-#ifdef BOOST_HAS_THREADS
```

Comment:
This include is still needed when neither <atomic> not <boost/atomic.hpp> provide the functionality we require.